### PR TITLE
client.d: is_watching() membership check optimization

### PR DESCRIPTION
**Pull request by slook**
- Changed: Avoid returning entire list of `users` objects just to do a `username` membership check
- Removed: Pointless iteration through `users` objects to check if iterating through `users` objects is going to happen
- Removed: `client.watched_by()` method, since it was only called (twice) by a single function
- Removed: `server.find_user()` method, since no function called it
- Added: `room.is_joined()` method for membership checking without exposing the entire `room.user_list`

Reduces unnecessary recursive iteration by several orders of magnitude.

### CODE REVIEW

File: src/client.d [hunk 1]
```diff
@@ -278,44 +278,26 @@ class User
...
-		foreach (username ; watch_list) {
-			auto user = server.get_user(username);
-			if (user) list[username] = user;
-		}
+		if (peer_username in watch_list)
+			return true;
 
 		foreach (room_name, room ; joined_rooms)
-			foreach (User user ; room.users) list[user.username] = user;
+			foreach (User user ; room.users) if (user.username == peer_username)
```
Comment (mathiascode) [inline src/client.d:287]: We can use room.user_names instead, check membership there.

Comment (slook) [reply src/client.d:291]: I've tried doing this with both `in` or `canFind()` but I don't know how to make it work...

----
`		 	if (peer_username in room.user_names)`
```
src/client.d(291): Error: incompatible types for `(peer_username) in (room.user_names())`: `string` and `string[]`
```

----
`		 	if (canFind(peer_username, room.user_names))`
```
src/client.d(292): Error: none of the overloads of template `std.algorithm.searching.canFind` are callable using argument types `!()(string, string[])`
```
----

Obvs I made `rooms.usernames` non-`private` but I think the array is not hashable or something like that, idk.

Comment (slook) [reply src/client.d]: Oh I see what you mean by checking it in there as in new method in the room module itself. Done!

Comment (mathiascode) [reply src/client.d]: Your first attempt is what I initially had in mind, but I'm too used to Python and wasn't aware it doesn't work. The new method is cleaner anyway.

**Pull request merged**